precise_driver/src/precise_hw_interface.cpp

```diff
@@ -144,18 +145,36 @@ namespace precise_driver
 
         if(req.data)
         {
-            res.success = resetController(false) && device_->freeMode(req.data);
+            res.success = resetController(false) && device_->freeMode(req.data, 31);
         }
         else
         {
-            res.success = resetController(true) && device_->freeMode(req.data);
+            res.success = resetController(true) && device_->freeMode(req.data, 31);
         }
 
         enableWrite(true);
 
         return true;
     }
 
+    bool PreciseHWInterface::teachmodeGrippedCb(std_srvs::SetBool::Request &req, std_srvs::SetBool::Response &res)
+    {
+        enableWrite(false);
+
+        if(req.data)
+        {
+            res.success = resetController(false) && device_->freeMode(req.data, 15);
+        }
+        else
+        {
+            res.success = resetController(true) && device_->freeMode(req.data, 15);
+        }
+
+        enableWrite(true);
+
+        return true;
+    }
```

Comment:
why not change the service message from `SetBool` to e.g. `SetInt` or even an precise-specific `SetFreeMode` service (that has a bool and an int in the request) to specify the axis in the request instead of duplicating the service just to call `freeMode` with another hardcoded bitmask/int?

Reply:
This was also my intention but it is not backward compatible. If @destogl agrees to it I can change this.

Reply:
I dont think we need to pay attention too much on backwards compatibility here....as both robots (daichii and böhringer) will be in Leinfelden within the next two weeks and we'll make it work for those....ipa kevin will eventually get the same update at some point, too

Reply:
Should be done but was not testet yet.

Reply:
Was tested and works.